java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -623,7 +518,7 @@ private MethodTreeImpl newMethodOrConstructor(
       throwsClauses = throwsClause.get().second();
     }
 
-    MethodTreeImpl result = new MethodTreeImpl(
+    return new MethodTreeImpl(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)